doctr/datasets/svhn.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import h5py
+import numpy as np
+from tqdm import tqdm
+
+from .datasets import VisionDataset
+
+__all__ = ['SVHN']
+
+
+class SVHN(VisionDataset):
+    """SVHN dataset from `"The Street View House Numbers (SVHN) Dataset"
+    <http://ufldl.stanford.edu/housenumbers/>`_.
+
+    Example::
+        >>> from doctr.datasets import SVHN
+        >>> train_set = SVHN(train=True, download=True)
+        >>> img, target = train_set[0]
+
+    Args:
+        train: whether the subset should be the training one
+        sample_transforms: composable transformations that will be applied to each image
+        rotated_bbox: whether polygons should be considered as rotated bounding box (instead of straight ones)
+        **kwargs: keyword arguments from `VisionDataset`.
+    """
+    TRAIN = ('http://ufldl.stanford.edu/housenumbers/train.tar.gz',
+             '4b17bb33b6cd8f963493168f80143da956f28ec406cc12f8e5745a9f91a51898',
+             'svhn_train.tar')
+
+    TEST = ('http://ufldl.stanford.edu/housenumbers/test.tar.gz',
+            '57ac9ceb530e4aa85b55d991be8fc49c695b3d71c6f6a88afea86549efde7fb5',
+            'svhn_test.tar')
+
+    MOCK_PATH = ''
+
+    def __init__(
+        self,
+        train: bool = True,
+        sample_transforms: Optional[Callable[[Any], Any]] = None,
+        rotated_bbox: bool = False,
+        **kwargs: Any,
+    ) -> None:
+
+        if self.MOCK_PATH:
+            self.root = self.MOCK_PATH
+        else:
+            url, sha256, name = self.TRAIN if train else self.TEST
+            super().__init__(url=url, file_name=name, file_hash=sha256, extract_archive=True, **kwargs)
```

Comment:
@fg-mindee 
not really happy with this i think you too :sweat_smile: 
suggestions ?

Reply:
No need for MOCK_PATH, we'll monkeypatch the `TRAIN` & `TEST` attributes :+1: